### ISSUE TITLE
AUR Package

### ISSUE THREAD

**Issue by lucazade** [Zren/material-decoration#14]
I would be nice to have an AUR package for ArchLinux and Manjaro

**Comment by zzag** [comment]
>  I should probably do that too, and maybe rename the project to differentiate from @zzag's (archived) project.

Yeah, it's a good idea. I wonder if it's possible to move the ownership of a repo to somebody else so the fork becomes "a new upstream."

**Comment by Zren** [comment]
* [ ] Confirm `Id` doesn't matter for cmake install
* [ ] Confirm behavior when `library` in `kwinrc` is invalid.
* [ ] Change `Id`
* [ ] Write upgrade process in `ReadMe`

----

I was reluctant to change the id without changing the name as existing users may end up with 2 installed versions of it, requiring notifying everyone to do this:

```
git checkout v0
cd build
sudo make uninstall
git checkout master
# build + install...
```

However since the `.so` file is overwriten, that not a problem.

----

It doesn't look like the `Id` matters for the installation, as it's installed to:

```
/usr/lib/qt/plugins/org.kde.kdecoration2/materialdecoration.so
```

So changing the Id will overwrite the existing `*.so` file. So the user will not end up with 2 installed version of the kdecoration (I need to confirm this after supper).

The main issue with existing users is the need to switch to Breeze first, then reinstall Material, then select Material again. As they currently have:
```
# ~/.config/kwinrc
[org.kde.kdecoration2]
library=com.github.zzag.material
theme=Material
```

I'm not sure if Plastic or Breeze is used if `kwinrc` has an invalid `library`.

**Comment by zzag** [comment]
> I'm not sure if Plastic or Breeze is used if kwinrc has an invalid library.

It depends on whether Breeze is available at compile-time. If it is, kwin will fallback to breeze; otherwise it will fallback to plastik.

**Comment by lucazade** [comment]
thanks for the ongoing work.. any update on this? :)

**Comment by Zren** [comment]
https://aur.archlinux.org/packages/material-kwin-decoration-git/ Tell me if there's any issue.

I used `gnome-shell-git` as a base for creating a pretty `pkgver`. Used a few other tricks from `fluent-decoration-git` too.

* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=gnome-shell-git
* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=fluent-decoration-git

**Comment by zzag** [comment]
You don't need to specify -DCMAKE_BUILD_TYPE and -DCMAKE_INSTALL_PREFIX, makepkg will pick up the right ones.

You could also get rid of prepare() by specifying the build directory in the cmake command, e.g. cmake -B builddir -S sourcedir -Dfoobar, see https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=plasma5-wallpapers-dynamic

**Comment by Zren** [comment]
Needed to add `extra-cmake-modules` (mentioned in a comment on AUR). I've removed `prepare()` to use `cmake -B "build"` and `make -C "build"`. I'll keep `cmake --build` and `cmake --install` in mind, as this is the first I've heard of them.

**Comment by lucazade** [comment]
Thanks for the package.. works great!
I just add the prepare section in order to fix some params in src (please make them configurable!)

```
prepare() {
  cd "${srcdir}/${_pkgname}"
  git submodule init
  git submodule update

  sed -i 's/s_titleBarOpacityActive = 0.75/s_titleBarOpacityActive = 1.0/g' ${srcdir}/material-kwin-decoration/src/Decoration.cc
  sed -i 's/s_titleBarOpacityInactive = 0.85/s_titleBarOpacityInactive = 1.0/g' ${srcdir}/material-kwin-decoration/src/Decoration.cc
  sed -i 's/"blur": true/"blur": false/g' ${srcdir}/material-kwin-decoration/src/material.json
}
```
